### PR TITLE
Bearer token auth support

### PR DESCRIPTION
**Pull request by allmightyspiff**
Adds a `HttpBearerCredentials` class to support using a Bearer token for API calls
Added code coverage report generation
Changed the examples to support either Bearer or username/apikey

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -40,33 +40,45 @@ additions to the SoftLayer API.
...
 ```
 
 ### Creating a Client
 
 All clients are instances of `ApiClient`. Currently there is only one implementation, the `RestApiClient`. Simply
 instantiate it and provide your credentials:
 
+
+#### Username and API Key
+For using a Classic Infrastructure or IBM Cloud API key. When using the IBM Cloud Api key, your username is litterally `apikey`, more information about that can be found on the SLDN [Authenticating to the SoftLayer API](https://sldn.softlayer.com/article/authenticating-softlayer-api/#cloud-api) article.
```
Comment (camporter) [inline README.md:66]: `is the literal string`

File: src/main/java/com/softlayer/api/http/HttpBearerCredentials.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package com.softlayer.api.http;
+
+/** HTTP bearer authorization support for bearer token fromhttps://iam.cloud.ibm.com/identity/token */
+public class HttpBearerCredentials implements HttpCredentials {
+
+    public final String token;
```
Comment (camporter) [inline src/main/java/com/softlayer/api/http/HttpBearerCredentials.java:6]: Might be good to mark this protected.

File: README.md [hunk 3]
```diff
@@ -40,33 +40,45 @@ additions to the SoftLayer API.
...
+#### Username and API Key
+For using a Classic Infrastructure or IBM Cloud API key. When using the IBM Cloud Api key, your username is the literal string `apikey`, more information about that can be found on the SLDN [Authenticating to the SoftLayer API](https://sldn.softlayer.com/article/authenticating-softlayer-api/#cloud-api) article.
+
 ```java
 import com.softlayer.api.*;
 
 ApiClient client = new RestApiClient().withCredentials("my user", "my api key");
 ```
 
+#### Acesses Token
```
Comment (camporter) [inline README.md:74]: Access*

**Comment by camporter** [comment]
Can we have the version updated in `gen/pom.xml` and `examples/pom.xml` as well?

File: README.md [hunk 4]
```diff
@@ -296,4 +308,4 @@ fully qualified class name of your implementation on a single line in a file in
 
 ## Copyright
 
-This software is Copyright (c) 2020 The SoftLayer Developer Network. See the bundled LICENSE file for more information.
+This software is Copyright (c) 2021 The SoftLayer Developer Network. See the bundled LICENSE file for more information.
```
Comment (camporter) [inline README.md:311]: `LICENSE` can be updated as well.

File: src/main/java/com/softlayer/api/http/HttpCredentials.java [hunk 5]
```diff
@@ -2,4 +2,5 @@
 
 /** Base interface for all accepted HTTP credentials */
 public interface HttpCredentials {
+    public String getHeader();
```
Comment (camporter) [inline src/main/java/com/softlayer/api/http/HttpCredentials.java:5]: nit, `public` is implied here and can be removed.

**Review by camporter — APPROVED** [review]

File: src/main/java/com/softlayer/api/http/HttpBearerCredentials.java [hunk 6]
```diff
@@ -0,0 +1,20 @@
+package com.softlayer.api.http;
+
+/** HTTP bearer authorization support for bearer token fromhttps://iam.cloud.ibm.com/identity/token */
```
Comment (FernandoOjeda) [inline src/main/java/com/softlayer/api/http/HttpBearerCredentials.java]: from https://iam.cloud.ibm.com/identity/token */

**Review by FernandoOjeda — APPROVED** [review]
It is working well and it is looking good for me.

**Pull request merged**